src/test/UniswapV2LockstakeCallee.t.sol

```diff
@@ -0,0 +1,418 @@
+            address(callee),    // Receiver of collateral and external call address
+            flashData           // Data to pass in external call; if length 0, no call is done
+        );
+
+        // Partially take auction with callee
+        flashData = abi.encode(
+            address(buyer1),   // Address of the user (where profits are sent)
+            expectedProfit,    // Minimum DAI/NST profit [wad]
+            path               // Uniswap v2 path
+        );
+        vm.prank(buyer1); clip.take(
+            id,                // Auction id
+            20_000 * 10**18,   // Upper limit on amount of collateral to buy  [wad]
+            type(uint256).max, // Maximum acceptable price (DAI / collateral) [ray]
+            address(callee),   // Receiver of collateral and external call address
+            flashData          // Data to pass in external call; if length 0, no call is done
+        );
+        assertEq(mkr.balanceOf(address(callee)), 0, "invalid-callee-mkr-balance");
+        assertEq(dst.balanceOf(address(callee)), 0, "invalid-callee-dst-balance");
+        assertEq(mkr.balanceOf(buyer1), 0, "invalid-final-buyer2-mkr-balance");
+        assertEq(dst.balanceOf(buyer1), expectedProfit, "invalid-final-buyer2-dst-balance");
+
+        // Setup different buyer to take the rest of the auction
+        address buyer2 = address(222);
+        vm.prank(buyer2); dss.vat.hope(address(clip));
+        assertEq(mkr.balanceOf(buyer2), 0, "unexpected-initial-buyer1-mkr-balance");
+        assertEq(dst.balanceOf(buyer2), 0, "unexpected-initial-buyer1-dst-balance");
+        address profitAddress = address(333);
+
+        // Take the rest of the auction with callee
+        expectedProfit = (12_000 * 10**18) * exchangeRate - 12_000 * pip.read() * clip.buf() / RAY;
+        flashData = abi.encode(
+            address(profitAddress), // Address of the user (where profits are sent)
+            expectedProfit,      // Minimum dai profit [wad]
+            path                    // Uniswap v2 path
+        );
+        vm.prank(buyer2); clip.take(
+            id,                // Auction id
+            type(uint256).max, // Upper limit on amount of collateral to buy  [wad]
+            type(uint256).max, // Maximum acceptable price (DAI / collateral) [ray]
+            address(callee),   // Receiver of collateral and external call address
+            flashData          // Data to pass in external call; if length 0, no call is done
+        );
+        assertEq(mkr.balanceOf(address(callee)), 0, "invalid-callee-mkr-balance");
+        assertEq(dst.balanceOf(address(callee)), 0, "invalid-callee-dst-balance");
+        assertEq(mkr.balanceOf(buyer2), 0, "invalid-final-buyer2-mkr-balance");
+        assertEq(dst.balanceOf(profitAddress), expectedProfit, "invalid-final-profit");
+    }
+
+    // --- Callee tests using MKR/DAI path ---
+
+    function testCalleeTakeNoWithStakingNoDelegateMkrDai() public {
```

Comment:
nit,
- Could we use `with` / `without` for unified function names?
eg. `testCalleeTakeWithoutStakingWithoutDelegateMkrDai` /`testCalleeTakeWithStakingWithDelegateMkrDai`
- For me, it'd be easier to read the test if the test function name sequence matches with `_testCalleeTake` params sequence.
eg. `testCalleeTakeWithoutDelegateWithoutStakingMkrDai`